### PR TITLE
7658 - working on the backend for matching user

### PR DESCRIPTION
**Pull request by codyseibert**

### CODE REVIEW

File: shared/src/authorization/authorizationClientService.js [hunk 1]
```diff
@@ -3,6 +3,7 @@ const ROLE_PERMISSIONS = {
   ADD_EDIT_JUDGE_USER: 'ADD_EDIT_JUDGE_USER',
   ADD_EDIT_PRACTITIONER_USER: 'ADD_EDIT_PRACTITIONER_USER',
   ADD_EDIT_STATISTICS: 'ADD_EDIT_STATISTICS',
+  ADD_PETITIONER_TO_CASE: 'ADD_PETITIONER_TO_CASE',
```
Comment (matthopson) [inline shared/src/authorization/authorizationClientService.js:6]: We have an `ASSOCIATE_USER_WITH_CASE` permission. Not sure if that does what you want here or not.

File: shared/src/business/useCases/addExistingUserToCaseInteractor.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  if (contactPrimary && contactPrimary.name === name) {
+    contactPrimary.serviceIndicator = SERVICE_INDICATOR_TYPES.SI_ELECTRONIC;
+    contactPrimary.email = email;
+    updatedContact = true;
+  }
+
+  const { contactSecondary } = caseEntity;
+  if (contactSecondary && contactSecondary.name === name) {
+    contactSecondary.serviceIndicator = SERVICE_INDICATOR_TYPES.SI_ELECTRONIC;
+    contactSecondary.email = email;
```
Comment (matthopson) [inline shared/src/business/useCases/addExistingUserToCaseInteractor.js:60]: It looks like we're going to be bale to add a separate email for the contactSecondary, so we may end up changing this.

File: web-api/src/cases/addExistingUserToCaseLambda.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+const { genericHandler } = require('../genericHandler');
+
+/**
+ * updates other statistics on the case (litigation costs and damages)
```
Comment (matthopson) [inline web-api/src/cases/addExistingUserToCaseLambda.js]: jsdoc

File: web-client/src/presenter/actions/CaseDetail/addExistingUserToCaseAction.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+ * @param {object} providers.props the cerebral props object
+ * @param {Function} providers.get the cerebral get helper function
+ * @returns {Promise} async action
+ */
+export const addExistingUserToCaseAction = async ({
+  applicationContext,
+  get,
+}) => {
+  const { docketNumber } = get(state.caseDetail);
+  const email = get(state.form.email) || 'petitioner1@example.com'; // TODO: remove fallback
```
Comment (matthopson) [inline web-client/src/presenter/actions/CaseDetail/addExistingUserToCaseAction.js:16]: We'll probably need to use `contactPrimary.email` and `contactSecondary.email` - I'm working through some of those logistics with validation, so perhaps something we come back and decide on later?

Comment (codyseibert) [reply web-client/src/presenter/actions/CaseDetail/addExistingUserToCaseAction.js:16]: yeah let's address this when we integrate the things together

File: shared/src/business/useCases/addExistingUserToCaseInteractor.js (hunk 2 above)
Comment (codyseibert) [reply shared/src/business/useCases/addExistingUserToCaseInteractor.js:60]: ok, no worries, we can change it as needed

File: shared/src/authorization/authorizationClientService.js (hunk 1 above)
Comment (codyseibert) [reply shared/src/authorization/authorizationClientService.js:6]: ASSOCIATE_USER_WITH_CASE seems to be available for all internal users, so I went with a new permission that only this admissions clerk should have at this point.

**Pull request merged**